File: LeetCode/1695. Maximum Erasure Value/Solution.py
class Solution:
    def makeFancyString(self, s: str)->:
        if (len(s) < 3):
            return s
        result = list(s[:2])
        for i in range(2, len(s)):
            if s[i] == s[i - 1] == s[i - 2]:
                continue
            result.append(s[i])
        
        return ''.join(result)

File: LeetCode/1695. Maximum Erasure Value/Solution.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
    string makeFancyString(string s) {
        int n = s.length();
        if (n <= 2) {return s};

        int write = 2;

        for(int read = 2; read < n; ++read) {
            if(s[read] == s[write - 1] && s[read] == s[write - 2]) {
                continue;
            }
            s[write] = s[read];
            write++;
        }

        s.resize(write);
        return s;
    }
}